### PR TITLE
core: --help option added

### PR DESCRIPTION
**Pull request by MathWay**

### CODE REVIEW

File: main.c [hunk 1]
```diff
@@ -156,3 +162,15 @@ int main(int _argc, char** _argv)
 	stop_code(EX_OK, NULL);
 }
 
+void help(void)
```
Comment (pfactum) [inline main.c]: Should be "static".

File: main.c [hunk 2]
```diff
@@ -39,6 +39,8 @@
 #include "utils.h"
 #include "worker.h"
 
+void help(void) __attribute__((noreturn));
```
Comment (pfactum) [inline main.c]: Do we really need forward declaration instead of just moving help() on the top?

File: main.c [hunk 3]
```diff
@@ -156,3 +162,15 @@ int main(int _argc, char** _argv)
 	stop_code(EX_OK, NULL);
 }
 
+void help(void)
+{
+	inform("Usage: dnsbalancer --help | --config=<path> [--pid-file=<path>] [--daemonize]\n", NULL);
```
Comment (pfactum) [inline main.c]: "--pid-file" is no longer relevant since 9005caa4c773ab582348a17369a39da4bd5e7373.

Comment (pfactum) [inline main.c]: Proper inform usage for pure text is: `inform("%s\n", "text");`

File: main.c [hunk 4]
```diff
@@ -156,3 +162,15 @@ int main(int _argc, char** _argv)
...
+	inform("                   [--verbose] [--debug] [--syslog]\n\n", NULL);
+	inform("  --config=<path>    specifies configuration file to use (mandatory)\n", NULL);
```
Comment (pfactum) [inline main.c]: TAB (`\t`) for indent?

File: main.c (hunk 2 above)
Comment (MathWay) [inline main.c]: No, it's just a matter of coding style. Used to keep main() on top.

File: main.c [hunk 5]
```diff
@@ -39,6 +39,23 @@
...
+	message("Usage: dnsbalancer --help | --config=<path> [--daemonize] [--verbose] [--debug]");
+	message("                   [--syslog]");
+	message("");
+	message("  --config=<path>    specifies configuration file to use (mandatory)");
+	message("  --daemonize        enables daemonization (preferred way to run on server)");
+	message("  --verbose          enables verbose output");
+	message("  --debug            enables debug output");
+	message("                     works only if compiled with MODE=DEBUG, otherwise does nothing");
+	message("  --syslog           logs everything to syslog instead of /dev/stderr");
+	message("  --help             shows this help and exit");
```
Comment (pfactum) [inline main.c]: "exits"?

File: pfcq.h [hunk 6]
```diff
@@ -35,6 +35,7 @@
 #define __noop					(void)0
 
 #define inform(A, ...)			__pfcq_debug(1, A, __VA_ARGS__)
+#define message(A)				inform("%s\n", A)
```
Comment (pfactum) [inline pfcq.h]: Shall we decouple `\n` from here, leaving only `%s` as format string? Suggestions?

File: main.c (hunk 5 above)
Comment (MathWay) [inline main.c]: Will be fixed.

File: pfcq.h (hunk 6 above)
Comment (MathWay) [inline pfcq.h]: I thought about this when was adding. IMHO, usage of message() looks cleaner this way. Usually you don't want to have multistring in one call, so it won't be confusing.

Comment (pfactum) [inline pfcq.h]: OK, ack.

File: main.c (hunk 5 above)
Comment (pfactum) [inline main.c]: OK. /me waits for resubmission.

**Pull request merged**